ICU-21580 Fix typos

luzpaz opened the pull request:
Various typos fixed. Sorry in advance for not following the contribution guidelines.  Can someone please help by opening a JIRA ticket for me. Having issues signing up and gave up. 

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21580
- [x] Required: The PR title must be prefixed with a JIRA Issue number. <!-- For example: "ICU-1234 Fix xyz" -->
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. <!-- For example: "ICU-1234 Fix xyz" -->
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

ALLOW_MANY_COMMITS=true

jefgen commented:
> Various typos fixed. Sorry in advance for not following the contribution guidelines. Can someone please help by opening a JIRA ticket for me. Having issues signing up and gave up.

Thanks for creating this PR!

There is an existing ticket for fixing typos and documentation for ICU 70.1 which could be used for this change. The ticket is 
ICU-21580.

Note: You'll need to update/change the commit message to start with the ICU ticket number, and then force-push the changes.

erik0686 commented:
Thank you very much for doing this work! We appreciate it a lot. Out of curiosity, did you use any tool to find or fix the typos? Or was it a manual search/replace?

luzpaz commented on icu4c/source/test/testdata/idna_conf.txt:
```diff
@@ -919,7 +919,7 @@ type: toascii
  nonbmp<10428><00ED>dn
 namebase: nonbmp<10400><00ED>dn
-nameutf8: nonbmp������ídn
+nameutf8: nonbmp������ídn
```
The fix I made to this file was on line 1236. Not sure what triggered the above issue, honestly.

```
desc: 5-1-2 Unassinged outside BMP; zone editors should reject
desc: 5-1-2 Unassigned outside BMP; zone editors should reject
```

jefgen replied:
Hmm.. perhaps the encoding of the file got changed when you saved it?
The file is UTF-8 with BOM. When I edit the file with VS Code it makes similar changes, but when I edit with Notepad ++ it does not.

If the only changes you made were on line 1236, then I think these need to be reverted.

erik0686 replied:
Hmmm might there have been some change in encoding when saving the file? @jefgen any ideas?

jefgen replied:
It looks like the character "𐐀" (U+10400) is encoded in UTF-8 in the file as a surrogate pair D801 DC00.
I think this is probably fine and/or perhaps intentional for the test case.

When the file is edited/saved it seems that the editor (or program that is changing the file) changes this to use the non-surrogate pair encoding, so it saves it as `F0 90 90 80` instead of `ed a0 81 ed b0 80 c3`.

I don't think we want to change the test case as part of fixing typos though. So I think these changes should be reverted in this file.

luzpaz replied:
I'm using gitkraken as my git client. For the life of me the only change I see is line 1236. I so no encoding changes.  
Screenshot:  
![image](https://user-images.githubusercontent.com/4140247/125700075-fd494ef9-3499-4d32-ab48-1b7cf57b5a12.png)

jefgen replied:
Hmm, IIRC, GitKraken is based on Electron, which is also what VS Code uses -- so that might explain why I see the same changes when I edit the file with VSCode.

What does git show in the terminal? When I edit the file with VSCode, `git diff` shows all the extraneous changes.

I wonder, maybe try reverting the changes to the file, and then using a different editor to change the one line, and then stage it using the terminal?

luzpaz commented:
@erik0686 I used [codespell](https://github.com/codespell-project/codespell) specifically `codespell -q 3 -L ans,anumber,atleast,ba,bre,hace,nd,nin,ois,rsource,som,sur,tht -S icu4c/source/data/zone,icu4c/source/data/lang`

I also ran some manual search and replace operations through my code editor.

@jefgen Thanks! I'll update the commit summary shortly to include `ICU-21580`

jefgen reviewed (changes requested):
Thanks for fixing all of this!

Please revert the changes in `icu4c\source\test\testdata\idna_conf.txt` other than line 1236.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/unicode-org/icu?pullRequest=1764) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/unicode-org/icu?pullRequest=1764) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: luzpaz<br/>:x: luz paz<br/><hr/>**luz paz** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/unicode-org/icu?pullRequest=1764) it.</sub>

jefgen commented:
Thanks for fixing the issues with the file `icu4c/source/test/testdata/idna_conf.txt`! 🙂

> **1** out of **2** committers have signed the CLA.

Hmm... Taking a look at the commits, it looks like the last commit had a different author -- which is causing the CLA bot to get confused.

Most of the commits are like this:
`From: luz paz <luzpaz@users.noreply.github.com>`
vs
`From: luz paz <luzpaz@github.com>`

There are a couple of ways you could try to change this:
- Since it is the last commit, you could use `git commit --amend --author "New Author Name <New Author Email>"` .
- Maybe try using the GitKraken client to change the author(?).
- Or you could try squashing the last two commits, and keep the author from the first commit.

However, the general practice for the ICU project is to squash commits before merging a PR, so may want to try squashing perhaps. (Though for this change, I wonder if it might be better to keep distinct commits for fixing typos under icu4c and icu4j).

luzpaz commented:
> (Though for this change, I wonder if it might be better to keep distinct commits for fixing typos under icu4c and icu4j).

Sorry for all the noise, folks. I've squashed all  icu4c commits down and separated from icu4j. So in total there are 2 commits.Also updated all commits so that email signature was uniform and re-signed the CLA. :crossed_fingers:

jefgen commented on icu4c/source/i18n/double-conversion-double-to-string.h:
```diff
@@ -138,7 +138,7 @@ class DoubleToStringConverter {
   // Example with max_leading_padding_zeroes_in_precision_mode = 6.
   //   ToPrecision(0.0000012345, 2) -> "0.0000012"
   //   ToPrecision(0.00000012345, 2) -> "1.2e-7"
-  // Similarily the converter may add up to
```
Note/FYI: The files that start with `double-conversion-*.h` actually come from an upstream project called ["double-conversion"](https://github.com/google/double-conversion). They are semi-regularly updated by pulling in the files from that project, which unfortunately means that this fix (and others) will eventually get overridden when they are updated... 

If you wanted to help contribute to another open-source project, (if you're so inclined) then you could maybe open a pull-request to fix them upstream. 😊

For example, this file is from here: https://github.com/google/double-conversion/blob/master/double-conversion/double-to-string.h#L126

luzpaz replied:
See https://github.com/google/double-conversion/pull/163 :wink:

jefgen replied:
Thank you very much!

jefgen reviewed (approved):
Thank you very much for fixing all of these typos! 👍 

I looked over all of the icu4c changes and LGTM.
@erik0686 can you look at the icu4j ones? Thanks!

jefgen commented:
> > (Though for this change, I wonder if it might be better to keep distinct commits for fixing typos under icu4c and icu4j).
> 
> Sorry for all the noise, folks. I've squashed all icu4c commits down and separated from icu4j. So in total there are 2 commits. Also updated all commits so that email signature was uniform and re-signed the CLA. 🤞

Thanks! It looks like the CLA bot is happier now! :)

erik0686 commented:
@jefgen Took a good look at icu4j and a quicker look at icu4c and it looks good to me :) :) It took a while but this was super worth it! thank you @luzpaz!!
